pkg/lib/cockpit-components-table.less

```diff
@@ -9,3 +9,45 @@ header {
 thead.listing-ct-empty tr td {
     font-weight: normal;
 }
+
+/* Fixes https://github.com/patternfly/patternfly-react/issues/3209 */
+table tbody.pf-m-expanded tr.pf-m-expanded th > div {
+    padding-top: var(--pf-c-table__expandable-row-content--PaddingTop);
+    padding-bottom: var(--pf-c-table__expandable-row-content--PaddingBottom);
+}
+
+/* Fix expandable button padding */
+.pf-c-table.pf-m-compact .pf-c-table__toggle {
+    --pf-c-table--m-compact-cell--first-last-child--PaddingLeft: 0;
+    margin-top: 0;
+}
+
+/* Align the expandable button icon */
+.pf-c-table__toggle svg {
+    vertical-align: middle;
+}
+
+/* Make expanded table lines extend to the content area */
+.pf-c-table tbody > tr.pf-m-expanded > :first-child::before {
+    background-color: var(--pf-c-table__expandable-row--before--BackgroundColor);
+}
+
+/* Remove line between row and row's expanded content */
```

Comment:
Generally against carrying and maintaining extra css just because patternfly upstream made an "incorrect" ascetics choice, but if you feel it's worth the burden, I'm fine with it. Might be worth a PR in the Patternfly issue tracker.